michael-eugene/assets/js/newsapi.js

```diff
@@ -0,0 +1,79 @@
+// BBC API
+var url = "https://newsapi.org/v2/top-headlines?sources=bbc-news&apiKey=f3d6aa29b4af4d45ad9cb6542b496c04";
+
+fetch(url)
+    .then((response) => response.json())
+    .then((response) => 
+
+        response.articles.slice(-4).forEach(art => {
+            document.getElementsByClassName('wrapper')[0].innerHTML += `
+        <div class="grid-item sm">
+            <div class="img-wrapper" style="background-image:url(${response.articles[0].urlToImage});"></div>
+            <div class="hero-txt">
+                <h2 class="grid-txt">${response.articles[0].title}</h2>
+                <p class="grid-txt">${response.articles[0].description}</p>
+                <div><a href="${response.articles[0].url}" class="btn-sec">Read More &rightarrow;</a></div>
+            </div>
+        </div>
+        `
+        }))
+
+// Technology API
+var url = "https://newsapi.org/v2/top-headlines?sources=techcrunch&apiKey=f3d6aa29b4af4d45ad9cb6542b496c04";
+
+fetch(url)
+    .then((response) => response.json())
+    .then((response) => 
+
+        response.articles.slice(-4).forEach(art => {
+            document.getElementsByClassName('wrapper')[1].innerHTML += `
+        <div class="grid-item sm">
+        <div class="img-wrapper" style="background-image:url(${response.articles[0].urlToImage});"></div>
+            <div class="hero-txt">
+                <h2 class="grid-txt">${response.articles[0].title}</h2>
+                <p class="grid-txt">${response.articles[0].description}</p>
+                <div><a href="${response.articles[0].url}" class="btn-sec">Read More &rightarrow;</a></div>
+            </div>
+        </div>
+        `
+        }))
+
+// Entertainment API
```

Comment:
Learn composition
 
You wrote almost the same thing for BBC, Technology and Entertainment. You could have done something like

```js
function fetchAndRenderNews(url, index) {
  fetch(url)
    .then((response) => response.json())
    .then((response) => 
         response.articles.slice(-4).forEach(art => {
            document.getElementsByClassName('wrapper')[index].innerHTML += `
        <div class="grid-item sm">
            <div class="img-wrapper" style="background-image:url(${response.articles[0].urlToImage});"> </div>
            <div class="hero-txt">
                <h2 class="grid-txt">${response.articles[0].title}</h2>
                <p class="grid-txt">${response.articles[0].description}</p>
                <div><a href="${response.articles[0].url}" class="btn-sec">Read More &rightarrow;</a></div>
            </div>
        </div>
        `
}
```